File: tests/Canaan.Tests.Loggers/Serilog.fs
namespace Canaan.Tests

open System

open Xunit

open Canaan

module SerilogTests = 

    [<Fact>]
    let ``Can create SerilogLogger ``() =
        printfn "%i log file(s) deleted." <| IO.DeleteFiles true "." "*.log"
        let s = SerilogLogger()
        do s.Info "Hello"
        IO.GetDirFiles "*.log" "." |> fst > 0 |> Assert.True
    

File: tests/Canaan.Tests.Loggers/Serilog.cs
using System;
using Xunit;

namespace Canaan.Tests
{
    public class SerilogTests
    {
        public SerilogTests()
        {
            Api.SetDefaultLoggerIfNone();
        }

        [Fact]
        public void CanCreateSerilogLogger()
        {
            IO.DeleteFiles("*.log");
            Assert.Equal(0, IO.GetFiles("*.log").Item1);
            Api.SetLogger(new SerilogLogger("Canaan.Tests.Loggers.log"));
            Assert.NotNull(Api.Logger);
            Api.Info("Hello World");
            Assert.NotEqual(0, IO.GetFiles("*.log").Item1);

        }
    }
}
